src/helpers/getLocaleFromRequest/index.js

```diff
@@ -11,12 +11,21 @@ function getAcceptedLanguage (request) {
   return acceptedLanguages[0]
 }
 
+function getQueryFromRequest (request) {
+  return function (param, key) {
+    return decodeURIComponent(get(request, `query.${key}`, '')) || param
+  }
+}
+
 function getLocaleFromRequest (options) {
   var cookieName = get(options, 'cookieName', 'locale')
+  var params = get(options, 'queryParams', ['locale'])
+  var queryParams = typeof params === 'string' ? [params] : params
 
   return function (request) {
+    var query = queryParams.reduce(getQueryFromRequest(request), null)
```

Comment:
Given this is not a `query`, I suggest we skip the variable and move this expression on line 28 directly.